src/utils/auth.ts

```diff
@@ -188,26 +210,24 @@ export default class Auth {
     })
   }
 
-  setSession(authResult: OAuthClient.Token) {
+  setSession(accessToken: string, expiresInSeconds: string) {
     return new Promise((resolve, reject) => {
       if (typeof localStorage === 'undefined') {
         reject('localStorage is not defined')
       }
-      const expires: unknown = authResult.data.expires_in
-      const expiresAt = JSON.stringify(
-        (expires as number) * 1000 + new Date().getTime(),
-      )
       const now: number = new Date().getTime()
 
-      const expiresInDays = Math.floor(
-        (Number(expiresAt) - now) / (60 * 60 * 24 * 1000),
-      )
+      const millisecondsInASecond = 1000
+      const expiresAt = Number(expiresInSeconds) * millisecondsInASecond + now
+
+      const millisecondsInADay = 60 * 60 * 24 * 1000
+      const expiresInDays = Math.floor((expiresAt - now) / millisecondsInADay)
```

Comment:
I struggled to figure out what was going on in this function and what some of the numbers were for. Once I made sense of it, I encoded my findings in variable names.